deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -5,371 +5,156 @@
+     * @return the created IotHubSSLContext.
+     * @throws CertificateException if the provided trustedCertificates are not PEM formatted and cannot be parsed.
+     * @throws IOException if the provided trustedCertificates cannot be read as a stream.
+     * @throws KeyStoreException if a key store cannot be created.
+     * @throws NoSuchAlgorithmException if your device cannot use x509 certificates or TLS SSLContexts.
+     * @throws KeyManagementException if the created SSLContext cannot be initialized.
      */
-    private TrustManagerFactory generateTrustManagerFactory(IotHubCertificateManager certificateManager, KeyStore trustKeyStore)
-            throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException
+    public static SSLContext getSSLContextFromString(String trustedCertificates) throws CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException, KeyManagementException
     {
-        if (trustKeyStore == null)
+        if (trustedCertificates == null || trustedCertificates.isEmpty())
         {
-            trustKeyStore = KeyStore.getInstance(KeyStore.getDefaultType());
-            trustKeyStore.load(null);
+            throw new IllegalArgumentException("The provided certificate string cannot be null or empty");
         }
 
-        for (Certificate c : certificateManager.getCertificateCollection())
+        try (InputStream inputStream = new ByteArrayInputStream(trustedCertificates.getBytes()))
         {
-            trustKeyStore.setCertificateEntry(TRUSTED_IOT_HUB_CERT_PREFIX + UUID.randomUUID(), c);
+            return getSSLContextFromStream(inputStream);
         }
-
-        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-        trustManagerFactory.init(trustKeyStore);
-
-        return trustManagerFactory;
     }
 
-    private static char[] generateTemporaryPassword()
+    /**
+     * Create an IotHubSSLContext that trusts the PEM formatted certificates stored in a file with the provided path.
+     * @param trustedCertificatesFilePath the absolute file path of the file that contains the PEM formatted
+     * certificates that this IotHubSSLContext will trust.
+     * @return the created IotHubSSLContext.
+     * @throws CertificateException if the provided trustedCertificates are not PEM formatted and cannot be parsed.
+     * @throws IOException if the provided trustedCertificates cannot be read as a stream.
+     * @throws KeyStoreException if a key store cannot be created.
+     * @throws NoSuchAlgorithmException if your device cannot use x509 certificates or TLS SSLContexts.
+     * @throws KeyManagementException if the created SSLContext cannot be initialized.
+     */
+    public static SSLContext getSSLContextFromFile(String trustedCertificatesFilePath) throws CertificateException, IOException, KeyStoreException, NoSuchAlgorithmException, KeyManagementException
```

Comment:
While we aren't parsing private keys anymore, we do have some scenarios where we need to parse public certs specifically around Edgehub, and soon around DPS